samza-azure/src/main/java/org/apache/samza/system/eventhub/consumer/EventHubSystemConsumer.java

```diff
@@ -254,6 +256,13 @@ private EventHubClientManager createOrGetEventHubClientManagerForSSP(String stre
     return eventHubClientManager;
   }
 
+  private void boundedInitializeEventHubsManagers() {
+    if (!ShutdownUtil.boundedExecution(Collections.singletonList(() -> initializeEventHubsManagers()),
```

Comment:
Would have to rename `ShutdownUtil` to use it here.

Reply:
didn't need to, found a better way